src/intersections/lines.jl

```diff
@@ -9,20 +9,24 @@ The intersection type can be one of three types:
 2. overlap at more than one point
 3. do not overlap nor intersect
 =#
-function intersection(f, l1::Line{3,T}, l2::Line{3,T}) where {T}
+function intersection(f::Function, l1::Line{3,T}, l2::Line{3,T}) where {T}
   a, b = l1(0), l1(1)
   c, d = l2(0), l2(1)
 
-  if measure(Tetrahedron(a, b, c, d)) > 0
+  if measure(Tetrahedron(a, b, c, d)) > 0 # not in same plane
     return @IT NoIntersection nothing f
-  elseif isapprox(abs((b - a) × (c - d)), zero(T), atol=atol(T)^2)
-    return @IT OverlappingLines l1 f
+  elseif isapprox(norm((b - a) × (c - d)), zero(T), atol=atol(T)^2)
+    if a in l2
+      return @IT OverlappingLines l1 f
+    else # parallel lines
+      return @IT NoIntersection nothing f
+    end
   else
     return @IT CrossingLines intersectpoint(l1, l2) f
   end
 end
 
-function intersection(f, l1::Line{2,T}, l2::Line{2,T}) where {T}
+function intersection(f::Function, l1::Line{2,T}, l2::Line{2,T}) where {T}
```

Comment:
I added the type of `f`. 
Not sure if it was intended to keep it unspecified

Reply:
Yes, it was intended. We should remove type annotations when they don't help with multiple dispatch. The code is more flexible that way.

Reply:
This issue is not resolved, can you please clarify why you requested a new round of reviews when the code remains unchanged?

Reply:
Now I removed the type specification